sdk/src/main/java/com/horizen/box/ForgerBox.java

```diff
@@ -1,38 +1,45 @@
 package com.horizen.box;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.primitives.Bytes;
 import com.google.common.primitives.Longs;
+import com.horizen.box.data.ForgerBoxData;
+import com.horizen.box.data.ForgerBoxDataSerializer;
 import com.horizen.proposition.PublicKey25519Proposition;
 import com.horizen.vrf.VRFPublicKey;
 
 import java.util.Arrays;
 import java.util.Objects;
 
+import static com.horizen.box.CoreBoxesIdsEnum.ForgerBoxId;
+
 public final class ForgerBox
-        extends PublicKey25519NoncedBox<PublicKey25519Proposition>
+        extends AbstractNoncedBox<PublicKey25519Proposition, ForgerBoxData>
         implements CoinsBox<PublicKey25519Proposition>
 {
-    public static final byte BOX_TYPE_ID = 2;
 
-    private final PublicKey25519Proposition rewardProposition;
-    private final VRFPublicKey vrfPubKey;
+    public ForgerBox(ForgerBoxData boxData, long nonce) {
+        super(boxData, nonce);
+    }
 
-    public ForgerBox(PublicKey25519Proposition proposition, long nonce, long value,
-                     PublicKey25519Proposition rewardProposition, VRFPublicKey vrfPubKey)
-    {
-        super(new PublicKey25519Proposition(proposition.pubKeyBytes()), nonce, value);
+    @Override
+    public byte boxTypeId() {
+        return ForgerBoxId.id();
+    }
 
-        Objects.requireNonNull(rewardProposition, "rewardProposition shall be defined");
-        Objects.requireNonNull(vrfPubKey, "vrfPubKey shall be defined");
+    @JsonProperty("vrfPubKey")
+    public VRFPublicKey vrfPubKey() {
+        return boxData.vrfPublicKey();
+    }
 
-        this.rewardProposition = new PublicKey25519Proposition(rewardProposition.pubKeyBytes());
-        this.vrfPubKey = new VRFPublicKey(vrfPubKey.bytes());
+    @JsonProperty("rewardProposition")
+    public PublicKey25519Proposition rewardProposition() {
+        return boxData.rewardProposition();
     }
 
     @Override
     public byte[] bytes() {
-        return Bytes.concat(_proposition.bytes(), Longs.toByteArray(_nonce), Longs.toByteArray(_value),
-                rewardProposition.bytes(), vrfPubKey.bytes());
+        return Bytes.concat(Longs.toByteArray(nonce), ForgerBoxDataSerializer.getSerializer().toBytes(boxData));
```

Comment:
!!!!!! Forger box shall also include value for id calculation, otherwise, forger stake can't be verified during consensus verification!!!!